worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java

```diff
@@ -0,0 +1,160 @@
+    private CompoundTag compoundTag;
+
+    public PaperweightLazyCompoundTag(Supplier<net.minecraft.nbt.CompoundTag> compoundTagSupplier) {
+        super(new HashMap<>());
+        this.compoundTagSupplier = compoundTagSupplier;
+    }
+
+    public PaperweightLazyCompoundTag(net.minecraft.nbt.CompoundTag compoundTag) {
+        this(() -> compoundTag);
+    }
+
+    public net.minecraft.nbt.CompoundTag get() {
+        return compoundTagSupplier.get();
+    }
+
+    @Override
+    public Map<String, Tag> getValue() {
+        if (compoundTag == null) {
+            compoundTag = (CompoundTag) WorldEditPlugin.getInstance().getBukkitImplAdapter().toNative(compoundTagSupplier.get());
+        }
+        return compoundTag.getValue();
+    }
+
+    @Override
+    public CompoundBinaryTag asBinaryTag() {
+        getValue();
+        return compoundTag.asBinaryTag();
+    }
+
+    public boolean containsKey(String key) {
+        return compoundTagSupplier.get().contains(key);
+    }
+
+    public byte[] getByteArray(String key) {
+        return compoundTagSupplier.get().getByteArray(key);
+    }
+
+    public byte getByte(String key) {
+        return compoundTagSupplier.get().getByte(key);
+    }
+
+    public double getDouble(String key) {
+        return compoundTagSupplier.get().getDouble(key);
+    }
+
+    public double asDouble(String key) {
+        net.minecraft.nbt.Tag tag = compoundTagSupplier.get().get(key);
+        if (tag instanceof NumericTag) {
+            return ((NumericTag) tag).getAsDouble();
+        }
+        return 0;
+    }
+
+    public float getFloat(String key) {
+        return compoundTagSupplier.get().getFloat(key);
+    }
+
+    public int[] getIntArray(String key) {
+        return compoundTagSupplier.get().getIntArray(key);
+    }
+
+    public int getInt(String key) {
+        return compoundTagSupplier.get().getInt(key);
+    }
+
+    public int asInt(String key) {
+        net.minecraft.nbt.Tag tag = compoundTagSupplier.get().get(key);
+        if (tag instanceof NumericTag) {
+            return ((NumericTag) tag).getAsInt();
+        }
+        return 0;
+    }
+
+    public List<Tag> getList(String key) {
+        net.minecraft.nbt.Tag tag = compoundTagSupplier.get().get(key);
+        if (tag instanceof net.minecraft.nbt.ListTag) {
+            ArrayList<Tag> list = new ArrayList<>();
+            net.minecraft.nbt.ListTag nbtList = (net.minecraft.nbt.ListTag) tag;
+            for (net.minecraft.nbt.Tag elem : nbtList) {
+                if (elem instanceof net.minecraft.nbt.CompoundTag) {
+                    list.add(new PaperweightLazyCompoundTag((net.minecraft.nbt.CompoundTag) elem));
+                } else {
+                    list.add(WorldEditPlugin.getInstance().getBukkitImplAdapter().toNative(elem));
+                }
+            }
+            return list;
+        }
+        return Collections.emptyList();
+    }
+
+    //TODO 1.18 be a ListBinaryTag ?
```

Comment:
Converting the whole thing to CBT would require a lot of overhaul. Quite outside of the scope of this PR, and since upstream isn't moving heavily towards CBT, we shouldn't do that either too much.

Reply:
Can remove that then